go/server.go

```diff
@@ -349,42 +413,63 @@ func createLinkToken(c *gin.Context) {
 	c.JSON(http.StatusOK, gin.H{"link_token": linkToken})
 }
 
-type httpError struct {
-	errorCode int
-	error     string
+func convertCountryCodes(countryCodeStrs []string) []plaid.CountryCode {
+	countryCodes := []plaid.CountryCode{}
+
+	for _, countryCodeStr := range countryCodeStrs {
+		countryCodes = append(countryCodes, plaid.CountryCode(countryCodeStr))
+	}
+
+	return countryCodes
 }
 
-func (httpError *httpError) Error() string {
-	return httpError.error
+func convertProducts(productStrs []string) []plaid.Products {
+	products := []plaid.Products{}
+
+	for _, productStr := range productStrs {
+		products = append(products, plaid.Products(productStr))
+	}
+
+	return products
 }
 
 // linkTokenCreate creates a link token using the specified parameters
 func linkTokenCreate(
-	paymentInitiation *plaid.PaymentInitiation,
-) (string, *httpError) {
-	countryCodes := strings.Split(PLAID_COUNTRY_CODES, ",")
-	products := strings.Split(PLAID_PRODUCTS, ",")
+	paymentInitiation *plaid.LinkTokenCreateRequestPaymentInitiation,
+) (string, error) {
+	ctx := context.Background()
+	countryCodes := convertCountryCodes(strings.Split(PLAID_COUNTRY_CODES, ","))
+	products := convertProducts(strings.Split(PLAID_PRODUCTS, ","))
 	redirectURI := PLAID_REDIRECT_URI
-	configs := plaid.LinkTokenConfigs{
-		User: &plaid.LinkTokenUser{
-			// This should correspond to a unique id for the current user.
-			ClientUserID: "user-id",
-		},
-		ClientName:        "Plaid Quickstart",
-		Products:          products,
-		CountryCodes:      countryCodes,
-		Language:          "en",
-		RedirectUri:       redirectURI,
-		PaymentInitiation: paymentInitiation,
+
+	user := plaid.LinkTokenCreateRequestUser{
+		ClientUserId: time.Now().String(),
+	}
+
+	request := plaid.NewLinkTokenCreateRequest(
+		"Plaid Quickstart",
+		"en",
+		countryCodes,
+		user,
+	)
+
+	request.SetProducts(products)
+
+	if redirectURI != "" {
+		request.SetRedirectUri(redirectURI)
+	}
+
+	if paymentInitiation != nil {
+		request.SetPaymentInitiation(*paymentInitiation)
 	}
-	resp, err := client.CreateLinkToken(configs)
+
+	linkTokenCreateResp, _, err := client.PlaidApi.LinkTokenCreate(ctx).LinkTokenCreateRequest(*request).Execute()
+
 	if err != nil {
-		return "", &httpError{
-			errorCode: http.StatusBadRequest,
-			error:     err.Error(),
-		}
+		return "", err
 	}
-	return resp.LinkToken, nil
+
+	return linkTokenCreateResp.GetLinkToken(), nil
 }
 
 func assets(c *gin.Context) {
```

Comment:
so this isn't exactly true anymore i think. we should probably modify the error message we return if we don't want to support this.